lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -175,19 +157,82 @@ TruncatedDistribution * TruncatedDistribution::clone() const
   return new TruncatedDistribution(*this);
 }
 
+/* Get the simplified version (or clone the distribution) */
+Distribution TruncatedDistribution::getSimplifiedVersion() const
+{
+  // No simplification in the multivariate case for now
+  if (getDimension() > 1) return *this;
+  // Simplification of the 1D case
+  Distribution localDistribution(distribution_);
+  String kind(localDistribution.getImplementation()->getClassName());
+  Scalar lower = bounds_.getLowerBound()[0];
+  Scalar upper = bounds_.getUpperBound()[0];
+  // Delve into the antecedents until we get something not truncated
+  while (kind == "TruncatedDistribution")
+    {
+      const TruncatedDistribution * truncatedDistribution = static_cast<const TruncatedDistribution *>(localDistribution.getImplementation().get());
+      localDistribution = truncatedDistribution->getDistribution();
+      kind = localDistribution.getImplementation()->getClassName();
+      lower = std::max(lower, truncatedDistribution->getBounds().getLowerBound()[0]);
+      upper = std::min(upper, truncatedDistribution->getBounds().getUpperBound()[0]);
+    }
+  const Scalar a = localDistribution.getRange().getLowerBound()[0];
+  const Scalar b = localDistribution.getRange().getUpperBound()[0];
+  // If no truncation
+  const Scalar w = getWeight();
+  if ((lower <= a) && (upper >= b))
+    {
+      localDistribution.setWeight(w);
+      return localDistribution;
+    }
+  // Intersect the underlying distribution range and the truncation range
+  const Scalar alpha = std::max(lower, a);
+  const Scalar beta = std::min(upper, b);
+  // Actual simplifications
+  if (kind == "Uniform")
+    {
+      const Uniform * uniform(static_cast< const Uniform * >(localDistribution.getImplementation().get()));
+      Uniform simplified(alpha, beta);
+      simplified.setWeight(getWeight());
+      return simplified;
+    }
+  if (kind == "Normal")
+    {
+      const Normal * normal(static_cast< const Normal * >(localDistribution.getImplementation().get()));
```

Comment:
I'd rather use dynamic_cast instead of static_cast to enable the runtime check, just in case

Reply:
I hope there will be no "case": the introspection mechanism is at the heart of OT!